src/js/components/Select/stories/components/SearchInput.js

```diff
@@ -1,32 +1,30 @@
-import React, { createRef, Component } from 'react';
+import React, { useEffect, useRef } from 'react';
 
 import { TextInput } from '../../..';
 
 import { SearchBorderBox } from './SearchBorderBox';
 import { SearchInputContext } from './SearchInputContext';
 
-export class SearchInput extends Component {
-  textInputRef = createRef();
+export const SearchInput = props => {
+  const textInputRef = useRef();
 
-  componentDidMount() {
-    this.focusTimeout = setTimeout(() => {
-      this.textInputRef.current.focus();
+  useEffect(() => {
+    const focusTimeout = setTimeout(() => {
+      textInputRef.current.focus();
     }, 300);
-  }
 
-  componentWillUnmount() {
-    clearTimeout(this.focusTimeout);
-  }
+    return () => {
+      clearTimeout(focusTimeout);
+    };
+  });
```

Comment:
Aren't you missing the empty array to allow the function to run once when mounted and unmounted?

Reply:
Oops! I will fix it and submit again.